### PR TITLE
executor: add more cases about dynamic-mode with union/DML/subquery

### PR DESCRIPTION
**Pull request by qw4990**
### What problem does this PR solve?

Issue Number: related to #24150  

Problem Summary: executor: add more cases about dynamic-mode with union/DML/subquery

### Check List 

Tests 

- Unit test

### Release note 

- executor: add more cases about dynamic-mode with union/DML/subquery

### CODE REVIEW

**Review by rebelice — APPROVED** [review]
/lgtm

File: executor/partition_table_test.go [hunk 1]
```diff
@@ -949,6 +949,181 @@ func (s *partitionTableSuite) TestSplitRegion(c *C) {
 	tk.MustPartition(`select * from thash where a in (1, 10001, 20001)`, "p1").Sort().Check(result)
 }
 
+func (s *partitionTableSuite) TestDML(c *C) {
+	if israce.RaceEnabled {
```
Comment (tiancaiamao) [inline executor/partition_table_test.go:953]: B.T.W, we have so many parallel executors, apply, aggregation, partition table reader ... 
All the data race test is skipped?

**Review by tiancaiamao — COMMENTED** [review]
/LGTM

File: executor/partition_table_test.go [hunk 2]
```diff
@@ -1298,6 +1298,181 @@ func (s *partitionTableSuite) TestSplitRegion(c *C) {
...
+		c.Skip("exhaustive types test, skip race test")
+	}
+
+	tk := testkit.NewTestKitWithInit(c, s.store)
+	tk.MustExec("create database test_DML")
```
Comment (mjonss) [inline executor/partition_table_test.go:1307]: maybe add `defer tk.MustExec("DROP SCHEMA test_DML")` too?

File: executor/partition_table_test.go [hunk 3]
```diff
@@ -1298,6 +1298,181 @@ func (s *partitionTableSuite) TestSplitRegion(c *C) {
...
+}
+
+func (s *partitionTableSuite) TestUnion(c *C) {
+	if israce.RaceEnabled {
+		c.Skip("exhaustive types test, skip race test")
+	}
+
+	tk := testkit.NewTestKitWithInit(c, s.store)
+	tk.MustExec("create database test_union")
+	tk.MustExec("use test_union")
```
Comment (mjonss) [inline executor/partition_table_test.go:1368]: maybe add defer `tk.MustExec("DROP SCHEMA test_union")` too?

File: executor/partition_table_test.go [hunk 4]
```diff
@@ -1298,6 +1298,181 @@ func (s *partitionTableSuite) TestSplitRegion(c *C) {
...
+}
+
+func (s *partitionTableSuite) TestSubqueries(c *C) {
+	if israce.RaceEnabled {
+		c.Skip("exhaustive types test, skip race test")
+	}
+
+	tk := testkit.NewTestKitWithInit(c, s.store)
+	tk.MustExec("create database test_subquery")
+	tk.MustExec("use test_subquery")
```
Comment (mjonss) [inline executor/partition_table_test.go:1418]: maybe add defer `tk.MustExec("DROP SCHEMA test_subquery")` too?

**Review by mjonss — APPROVED** [review]
/LGTM (but I don't yet have reviewer status)
Only suggestion to drop the databases through defer after creating them.

**Comment by qw4990** [comment]
These conflicts are hard to resolve, so I create a new PR for this test cases https://github.com/pingcap/tidb/pull/24864;

**Pull request closed without merging**